src/datachain/lib/listing.py

```diff
@@ -173,7 +170,7 @@ def get_listing(
         uri = str(uri)
 
     # we don't want to use cached dataset (e.g. for a single file listing)
-    if not uri.endswith("/") and _isfile(client, uri):
+    if not glob.has_magic(uri) and not uri.endswith("/") and _isfile(client, uri):
```

Comment:
The `fs.info` does not support a glob. So, it should never go down that path, regardless of the platform.